src/Features/LanguageServer/Protocol/Handler/MethodAttribute.cs

```diff
@@ -4,15 +4,16 @@
 
 using System;
 using System.Composition;
+using CommonLanguageServerProtocol.Framework;
 
 namespace Microsoft.CodeAnalysis.LanguageServer.Handler
 {
     [MetadataAttribute]
-    [AttributeUsage(AttributeTargets.Class)]
-    internal class MethodAttribute : Attribute
+    [AttributeUsage(AttributeTargets.Class | AttributeTargets.Interface)]
+    public class MethodAttribute : Attribute
```

Comment:
I think it makes sense to have this in CommonLanguageServerProtocol, but not along with the `MetadataAttribute`. Unsure on what exactly to do. Might move this and then create a "MetadataMethodAttribute" which derives from the CLaSP one and  then adds the `MetadataAttribute` for Roslyn.

Reply:
Agree. Could even name it as `LSPHandler`, `RequestHandler`, `LanguageServerEndpoint` or something more specific to what its purpose is. The parameters to the attribute (method, metadata etc.) can add the color to what it actually means

Reply:
Oh, I like `[LanguageServerEndpoint(string method)]`. I was struggling with names because I was thinking of it as a generalized JsonRPC method but it's more specific than that.